scc/git.py

```diff
@@ -4033,3 +4033,183 @@ def __call__(self, args):
+            self.log.info("no new version")
+
+    def commit(self, msg):
+        p = subprocess.Popen(["git", "add", "-A"])
+        rc = p.wait()
+        if rc != 0:
+            raise Exception("'git add failed")
+        p = subprocess.Popen(["git", "status"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        output, error = p.communicate()
+        if b"nothing to commit" in output:
+            print("OK %s" % output)
+            self.log.info("Nothing to commit")
+            return
+        p = subprocess.Popen(["git", "commit", "-m", msg])
+        rc = p.wait()
+        if rc != 0:
+            raise Exception("'git commit failed")
+
+    def extact_metadata(self, directory, yaml):
+        """
+        Scan the directory and parse the content of the first
+        meta.yaml file found.
+        """
+        for (dirpath, dirnames, filenames) in os.walk(directory):
+            for fn in filenames:
+                if fn == self.META_FILE:
+                    fullpath = os.path.join(dirpath, fn)
+                    jinja2 = {}
+                    with open(fullpath) as fp:
+                        data = yaml.load(fp)
+                    # find information about the repository
+                    with open(fullpath) as file:
+                        for line in file:
+                            if line.startswith(self.PREFIX):
+                                values = self.replace(line)
+                                jinja2[values[0]] = values[1].replace("\"", "")
+                    return (data, jinja2)
+        return (None, None)
+
+    def replace(self, line):
+        """
+        Clean up the line
+        """
+        v = line.replace(self.PREFIX, "").replace(self.SUFFIX, "")
+        return v.replace(" ", "").replace("\n", "").split("=")
+
+    def get_latest_tag_and_sha256_from_github(self, repo_url):
+        """
+        Return the latest tag and the sha256 for the given repository.
+        """
+        command = "git ls-remote --refs --tags --sort=\"version:refname\" %s | grep -v 'v*.rc[0-9]\+$' | grep -v 'v*.dev[0-9]\+$' | grep 'v*\.' |grep -v 'v*.-m[0-9]\+$' | tail -n1" % repo_url  # noqa
+        process = subprocess.run(command, shell=True, check=True,
+                                 stdout=subprocess.PIPE,
+                                 universal_newlines=True)
+        return process.stdout.split("\t")
+
+    def get_sha256_from_pypi(self, repo_name, tag, extension=".tar.gz"):
+        """
+        Read the sha256 from pypi.
+        """
+        import requests
+        r = requests.get('https://pypi.org/pypi/%s/json' % repo_name)
+        json = r.json()
+        # parse the json to extract the sha256
+        for v in json["releases"][tag]:
+            if v["filename"].endswith(extension):
+                return v["digests"]["sha256"]
+
+    def get_sha256_from_downloads(self, data, tag, extension=".sha1"):
+        """
+        Read the sha256 from downloads.openmicroscopy.org.
```

Comment:
But the implementation will retrieve the SHA1 sum (as this is the one we compute at the moment), no?

Reply:
yes. I will rename variable so it is not confusing.